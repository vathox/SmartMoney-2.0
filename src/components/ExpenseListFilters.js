import React from 'react'
import {connect} from 'react-redux'
import {setTextFilter, sortByAmount, sortByDate, setStartDate, setEndDate} from '../actions/filters'
import {DateRangePicker} from 'react-dates'
import 'react-dates/initialize'

class ExpenseListFilters extends React.Component {
    state = {
        calendarFocused: null
    };

    onDatesChange = ({startDate, endDate}) => {
        this.props.dispatch(setStartDate(startDate));
        this.props.dispatch(setEndDate(endDate))
    };

    onFocusChange = (calendarFocused) => {
        this.setState(() => ({calendarFocused}))
    };

    render() {
        return (

            <div className="content-container">
                <div className="input_group">
                    <div className="input_group__item">
                        <input className="text_input"
                               placeholder="Search Expenses"
                               type="text"
                               value={this.props.filters.text}
                               onChange={(e) => {
                                   this.props.dispatch(setTextFilter(e.target.value))
                               }}/>
                    </div>
                    <div className="input_group__item">
                        <select className="select"
                                value={this.props.filters.sortBy}
                                onChange={(e) => {
                                    this.props.dispatch(e.target.value === "date" ? sortByDate() : sortByAmount())
                                }}>
                            <option value="date">Date</option>
                            <option value="amount">Amount</option>
                        </select>
                    </div>
                    <div className="input_group__item">
                        <DateRangePicker
                            startDate={this.props.filters.startDate}
                            endDate={this.props.filters.endDate}
                            onDatesChange={this.onDatesChange}
                            focusedInput={this.state.calendarFocused}
                            onFocusChange={this.onFocusChange}
                            showClearDates={true}
                            numberOfMonths={1}
                            isOutsideRange={() => false}
                        />
                    </div>
                </div>
            </div>
        )
    }
}


const mapStateToProps = (state) => {
    return {
        filters: state.filters
    }
};

export default connect(mapStateToProps)(ExpenseListFilters)