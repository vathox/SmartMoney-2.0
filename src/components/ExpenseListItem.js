import React from 'react'
import { Link } from 'react-router-dom'


const ExpenseListItem = ({id, description = "", amount = 0, createdAt = 0  }) =>(
    <div>
        <Link to={`/edit/${id}`}><h3>{description}</h3></Link>
        <p>Amount: {amount}</p>
        <p>Created at: {createdAt}</p>
    </div>
);


export default ExpenseListItem