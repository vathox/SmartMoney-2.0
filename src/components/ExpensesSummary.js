import React from 'react'
import {connect} from 'react-redux'
import selectExpensesTotal from '../selectors/expenses-total'
import selectExpenses from '../selectors/expenses'
import numeral from 'numeraljs';

export const ExpensesSummary = (props) => (
    <div>
        <p>Viewing {props.expenses.length} totalling {numeral(selectExpensesTotal(props.expenses)/100).format('$0,0.00')}</p>
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    }
};

export default connect(mapStateToProps)(ExpensesSummary)