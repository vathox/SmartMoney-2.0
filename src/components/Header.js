import React from 'react'
import {NavLink} from 'react-router-dom'

const Header = () => (
    <div>
        <h1>Smart Money 2.0</h1>
        <NavLink to="/" activeClassName="is-active" exact={true}>Dashboard</NavLink>
        <NavLink to="/create" activeClassName="is-active">Add new expense</NavLink>
        <NavLink to="/edit" activeClassName="is-active">Edit</NavLink>
        <NavLink to="/help" activeClassName="is-active">Help</NavLink>
    </div>
);

export default Header