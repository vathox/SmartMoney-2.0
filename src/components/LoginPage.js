import React from 'react'
import {connect} from 'react-redux'
import {startLogIn} from '../actions/auth'

const LoginPage = ({startLogIn}) =>(
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">SmartMoney 2.0</h1>
            <p>Take your expenses under control with smart solutions</p>
            <button className="button" onClick={startLogIn}>Log in with Google</button>
        </div>
    </div>
);

const mapDispatchToProps = (dispatch) =>({
    startLogIn : () => dispatch(startLogIn())
});

export default connect(undefined, mapDispatchToProps)(LoginPage)
