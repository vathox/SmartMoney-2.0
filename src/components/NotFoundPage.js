import React from 'react'

const NotFoundPage = () =>(
    <div>
        404 - <Link to="/"> Go to home page </Link>
    </div>
);

export default NotFoundPage