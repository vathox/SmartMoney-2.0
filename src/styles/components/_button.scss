.button {
  background: #4885ed;
  border: none;
  border-radius: 10px;
  color: white;
  display: inline-block;
  font-size: $font-size-large;
  font-weight: 500;
  line-height: 1;
  padding: $s-size;
  text-decoration: none;
}

.button--link{
  background: none;
}

.button--secondary{
  background: #ed2157;
}
.button--create{
  margin-top: $m-size;
}