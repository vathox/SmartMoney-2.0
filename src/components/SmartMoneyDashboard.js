import React from 'react'
import ExpenseList from './ExpenseList'
import ExpenseListFilters from './ExpenseListFilters'
import ExpensesSummary from './ExpensesSummary'

const SmartMoneyDashboard = () =>
    <div>
        <ExpensesSummary/>
        <ExpenseListFilters/>
        <ExpenseList/>
    </div>
;

export default SmartMoneyDashboard