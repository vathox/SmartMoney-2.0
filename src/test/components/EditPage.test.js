import React from 'react'
import {shallow} from 'enzyme'
import {EditPage} from "../../components/EditPage";
import expenses from '../fixtures/expenses'

let startEditExpense , startRemoveExpense, history, wrapper;

beforeEach(() => {
    startEditExpense = jest.fn();
    startRemoveExpense= jest.fn();
    history = {push: jest.fn()};
    wrapper = shallow(<EditPage
        startEditExpense={startEditExpense}
        history={history}
        startRemoveExpense={startRemoveExpense}
        expense={expenses[0]}/>)
});

test('should render edit page correctly', () => {
    expect(wrapper).toMatchSnapshot()
});

test('should handle start edit expense', () => {
    wrapper.find('ExpenseForm').prop('onSubmit')(expenses[0]);
    expect(history.push).toHaveBeenCalledWith('/');
    expect(startEditExpense).toHaveBeenCalledWith(expenses[0].id, expenses[0])
});

test('should handle remove expense', () => {
    wrapper.find('button').simulate('click');
    expect(history.push).toHaveBeenCalledWith('/');
    expect(startRemoveExpense   ).toHaveBeenCalledWith({id: expenses[0].id})
});