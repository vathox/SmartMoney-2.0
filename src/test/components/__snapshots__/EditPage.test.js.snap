// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render edit page correctly 1`] = `
<div>
  <ExpenseForm
    expense={
      Object {
        "amount": 193,
        "createdAt": 0,
        "description": "Car",
        "id": 1,
        "note": "Test note 1",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
